### PR TITLE
fix: load_model fallback to BytesIO for Py3.6

### LINKED ISSUE #163: (Py3.6) torch.load requires seekable file
**Describe the bug**

In Python < 3.7, `load_model()` fails because `ZipFile.open(...)` returns a *non-seekable* file-like object, which `torch.load()` will not accept. (See [this StackOverflow question](https://stackoverflow.com/q/33742544))

**What is the current behavior?**

Load model fails with:

```
 File "/opt/conda/lib/python3.6/site-packages/pytorch_tabnet/tab_model.py", line 308, in load_model
    saved_state_dict = torch.load(f)
  File "/opt/conda/lib/python3.6/site-packages/torch/serialization.py", line 525, in load
    with _open_file_like(f, 'rb') as opened_file:
  File "/opt/conda/lib/python3.6/site-packages/torch/serialization.py", line 217, in _open_file_like
    return _open_buffer_reader(name_or_buffer)
  File "/opt/conda/lib/python3.6/site-packages/torch/serialization.py", line 202, in __init__
    _check_seekable(buffer)
  File "/opt/conda/lib/python3.6/site-packages/torch/serialization.py", line 292, in _check_seekable
    raise_err_msg(["seek", "tell"], e)
  File "/opt/conda/lib/python3.6/site-packages/torch/serialization.py", line 285, in raise_err_msg
    raise type(e)(msg)
io.UnsupportedOperation: seek. You can only torch.load from a file that is seekable. Please pre-load the data into a buffer like io.BytesIO and try to load from it instead.
```

**If the current behavior is a bug, please provide the steps to reproduce.**

From discussion on the above linked SO post I believe `load_model()` will always fail on any Python <3.7 environment - but my specific configuration is provided below.

**Expected behavior**

Model saves and loads without error (I haven't seen any specific exclusions in the doc around Python & PyTorch version support?)

**Screenshots**

**Other relevant information:**
poetry version:  N/A
python version: 3.6
Operating System:  Linux (Ubuntu-based, per [the PyTorch v1.4 DLC doc](https://github.com/aws/deep-learning-containers/blob/master/available_images.md#prior-versions))
Additional tools:  Amazon SageMaker, PyTorch v1.4

**Additional context**

I could upgrade to PyTorch 1.5 but can't support Python 3.7+ in my environment just yet, so would really appreciate if we can keep compatibility!

Perhaps lightest solution would by an explicit `try/catch` inside `load_model` to buffer the .pt file to `BytesIO` memory if and only if `torch.load()` fails with `io.UnsupportedOperation`?

### PR DESCRIPTION
**Pull request by athewsey**
**IMPORTANT: Please do not create a Pull Request without creating an issue first.**

Proposed fix for #163 - failure of `load_model()` in Python <3.7 environments.

This is a pretty quick & dirty approach aiming to minimise footprint of the code change for compatibility, but there could be other alternatives:

- Could try to detect the situation (e.g. via Python version detection or determining whether the returned `f` object is seekable) rather than handling as an error catch, if that's preferred? I was just trying to minimise the situations in which the file would be buffered into memory to only those where it was strictly necessary.
- Could try [this approach](https://stackoverflow.com/a/54199495/13352657) of hacking around with `f` to make it **look** seekable and hoping that's enough - but this seemed sketchy to me as it could just mask implementation gaps into more subtle errors
- Maybe there are other workarounds that don't need to buffer the contents of the file into memory? I haven't found any so far but couldn't rule it out

Happy to discuss & revise, but thought a working suggestion would help things along :-)

**What kind of change does this PR introduce?** Bugfix

**Does this PR introduce a breaking change?** No

**What needs to be documented once your changes are merged?**

At least in my environment (Amazon SageMaker, SageMaker-provided PyTorch v1.4 container running Python 3.6) this fix appears sufficient to support training, saving, loading and executing a model with this Python/PyTorch version combination.

...Probably not worth explicitly documenting that without some more rigorous/automated environment testing though!

**Closing issues**

Closes #163

### CODE REVIEW

**Comment by eduardocarvp** [comment]
Thank you, @athewsey !

The fix seems pretty neat and the code changes are minimal. I just need to test it on different environments with older Python versions before validating, just letting you know that we're on it!

**Comment by Optimox** [comment]
As soon as your checks are OK @eduardocarvp it's ok for me!
